earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -40,30 +41,96 @@ ListElementTagName(const std::string &tagname)
 // This is used only in the following function
 class UsingXMLGuard
 {
-  friend void InitializeXMLLibrary(void) throw();
+  friend void InitializeXMLLibrary(bool) throw();
 
-  UsingXMLGuard(void) throw() {
-    try {
+  UsingXMLGuard(void) throw()
+  {
+   try {
       XMLPlatformUtils::Initialize();
     } catch(const XMLException& toCatch) {
       notify(NFY_FATAL, "Unable to initialize Xerces: %s",
              FromXMLStr(toCatch.getMessage()).c_str());
     }
   }
+
   ~UsingXMLGuard(void) throw() {
     try {
       XMLPlatformUtils::Terminate();
     } catch (...) {
     }
   }
+
+
+//  ------------------------------------------------------------------
+//  From: https://xml.apache.org/xerces-c-new/faq-parse.html#faq-4
+//
+//  Is it OK to call the XMLPlatformUtils::Initialize/Terminate pair
+//  of routines multiple times in one program?
+//
+//  Yes. Since Xerces-C++ 1.5.2, the code has been enhanced so that
+//  calling XMLPlatformUtils::Initialize/Terminate pair of routines
+//  multiple times in one process is now allowed.
+//
+//  But the application needs to guarantee that only one thread has
+//  entered either the method XMLPlatformUtils::Initialize() or the
+//  method XMLPlatformUtils::Terminate() at any one time.
+//
+//  If you are calling XMLPlatformUtils::Initialize() a number of
+//  times, and then follow with XMLPlatformUtils::Terminate() the same
+//  number of times, only the first XMLPlatformUtils::Initialize() will
+//  do the initialization, and only the last XMLPlatformUtils::Terminate()
+//  will clean up the memory. The other calls are ignored.
+//
+//  To ensure all the memory held by the parser are freed, the number of
+//  XMLPlatformUtils::Terminate() calls should match the number
+//  of XMLPlatformUtils::Initialize() calls.
+//  -------------------------------------------------------------------
+    void ReInitXerces(void) throw()
```

Comment:
I'd like to see the code for reinitializing xerces also moved to a separate WIP PR. I can see the point of keeping it around, but when we have code that is experimental and is known to cause segfaults, we shouldn't check it in to master.